src/Backend/AMQPBackendDispatcher.php

```diff
@@ -133,7 +133,7 @@ public function getChannel()
             ]);
 
             if ($factory instanceof DelayStrategyAware) {
-                $factory->setDelayStrategy(new RabbitMqDlxDelayStrategy());
+                $factory = $this->resolveDelayStrategy($factory);
```

Comment:
shouldn't we resolve this when instantiating the class?

cc @greg0ire

Reply:
Also, wouldn't all this be more simple if we were using DI with an optional service the user could define or not define? That way we don't have to instantiate things ourselves

https://symfony.com/doc/current/service_container/optional_dependencies.html